### ISSUE TITLE
Crash in loop-load-elim: Assertion `AR && "Invalid addrec expression"' failed.

### ISSUE THREAD

**Issue by efriedma-quic** [llvm/llvm-project#57368]
The following IR crashes with `opt -loop-load-elim`.  Regression from db8fcb2c2537.  https://godbolt.org/z/x3v4eqM4K .

(On a side-note, following IR is reduced with llvm-reduce, with very little editing by hand; my first time using it, and it was pleasant to use.)

```llvm
target datalayout = "e-m:e-i8:8:32-i16:16:32-i64:64-i128:128-n32:64-S128"

define void @f(double* %arrayidx3.i.i.us, double* %0) {
entry:
  br label %for.body.us

for.body.us:
  %indvars.iv.us1 = phi i64 [ %indvars.iv.next.us, %for.body.us ], [ 0, %entry ]
  %indvars79.us = trunc i64 %indvars.iv.us1 to i32
  store double 0.000000e+00, double* %arrayidx3.i.i.us, align 8
  %conv22.us = zext i32 %indvars79.us to i64
  %cond.in.v.us = select i1 true, i64 %conv22.us, i64 0
  %cond.in.us = getelementptr inbounds double, double* %0, i64 %cond.in.v.us
  %cond.us = load double, double* %cond.in.us, align 8
  %indvars.iv.next.us = add nuw nsw i64 %indvars.iv.us1, 1
  %exitcond.not.us = icmp eq i64 %indvars.iv.next.us, 0
  br i1 %exitcond.not.us, label %exit, label %for.body.us

exit:
  ret void
}
```

Crash trace:

```console
opt: /root/llvm-project/llvm/lib/Analysis/LoopAccessAnalysis.cpp:214: void llvm::RuntimePointerChecking::insert(llvm::Loop*, llvm::Value*, const llvm::SCEV*, llvm::Type*, bool, unsigned int, unsigned int, llvm::PredicatedScalarEvolution&, bool): Assertion `AR && "Invalid addrec expression"' failed.
PLEASE submit a bug report to https://github.com/llvm/llvm-project/issues/ and include the crash backtrace.
Stack dump:
0.	Program arguments: /opt/compiler-explorer/clang-assertions-trunk/bin/opt -o /app/output.s -S -loop-load-elim <source>
 #0 0x0000555d58777e74 PrintStackTraceSignalHandler(void*) Signals.cpp:0:0
 #1 0x0000555d58775634 SignalHandler(int) Signals.cpp:0:0
 #2 0x00007fa6fbef1420 __restore_rt (/lib/x86_64-linux-gnu/libpthread.so.0+0x14420)
 #3 0x00007fa6fb9be00b raise (/lib/x86_64-linux-gnu/libc.so.6+0x4300b)
 #4 0x00007fa6fb99d859 abort (/lib/x86_64-linux-gnu/libc.so.6+0x22859)
 #5 0x00007fa6fb99d729 (/lib/x86_64-linux-gnu/libc.so.6+0x22729)
 #6 0x00007fa6fb9aefd6 (/lib/x86_64-linux-gnu/libc.so.6+0x33fd6)
 #7 0x0000555d574d60ff llvm::RuntimePointerChecking::insert(llvm::Loop*, llvm::Value*, llvm::SCEV const*, llvm::Type*, bool, unsigned int, unsigned int, llvm::PredicatedScalarEvolution&, bool) (/opt/compiler-explorer/clang-assertions-trunk/bin/opt+0x25650ff)
 #8 0x0000555d574e3b46 (anonymous namespace)::AccessAnalysis::canCheckPtrAtRT(llvm::RuntimePointerChecking&, llvm::ScalarEvolution*, llvm::Loop*, llvm::DenseMap<llvm::Value const*, llvm::Value*, llvm::DenseMapInfo<llvm::Value const*, void>, llvm::detail::DenseMapPair<llvm::Value const*, llvm::Value*>> const&, llvm::Value*&, bool) (.part.0) LoopAccessAnalysis.cpp:0:0
 #9 0x0000555d574e546c llvm::LoopAccessInfo::analyzeLoop(llvm::AAResults*, llvm::LoopInfo*, llvm::TargetLibraryInfo const*, llvm::DominatorTree*) (/opt/compiler-explorer/clang-assertions-trunk/bin/opt+0x257446c)
#10 0x0000555d574e5dd8 llvm::LoopAccessAnalysis::run(llvm::Loop&, llvm::AnalysisManager<llvm::Loop, llvm::LoopStandardAnalysisResults&>&, llvm::LoopStandardAnalysisResults&) (/opt/compiler-explorer/clang-assertions-trunk/bin/opt+0x2574dd8)
#11 0x0000555d58b92068 llvm::detail::AnalysisPassModel<llvm::Loop, llvm::LoopAccessAnalysis, llvm::PreservedAnalyses, llvm::AnalysisManager<llvm::Loop, llvm::LoopStandardAnalysisResults&>::Invalidator, llvm::LoopStandardAnalysisResults&>::run(llvm::Loop&, llvm::AnalysisManager<llvm::Loop, llvm::LoopStandardAnalysisResults&>&, llvm::LoopStandardAnalysisResults&) (/opt/compiler-explorer/clang-assertions-trunk/bin/opt+0x3c21068)
#12 0x0000555d574e96bb llvm::AnalysisManager<llvm::Loop, llvm::LoopStandardAnalysisResults&>::getResultImpl(llvm::AnalysisKey*, llvm::Loop&, llvm::LoopStandardAnalysisResults&) (/opt/compiler-explorer/clang-assertions-trunk/bin/opt+0x25786bb)
#13 0x0000555d5849dc42 llvm::LoopAccessInfo const& llvm::function_ref<llvm::LoopAccessInfo const& (llvm::Loop&)>::callback_fn<llvm::LoopLoadEliminationPass::run(llvm::Function&, llvm::AnalysisManager<llvm::Function>&)::'lambda'(llvm::Loop&)>(long, llvm::Loop&) LoopLoadElimination.cpp:0:0
#14 0x0000555d584a48b1 eliminateLoadsAcrossLoops(llvm::Function&, llvm::LoopInfo&, llvm::DominatorTree&, llvm::BlockFrequencyInfo*, llvm::ProfileSummaryInfo*, llvm::ScalarEvolution*, llvm::AssumptionCache*, llvm::function_ref<llvm::LoopAccessInfo const& (llvm::Loop&)>) (.isra.0) LoopLoadElimination.cpp:0:0
#15 0x0000555d584a5a03 llvm::LoopLoadEliminationPass::run(llvm::Function&, llvm::AnalysisManager<llvm::Function>&) (/opt/compiler-explorer/clang-assertions-trunk/bin/opt+0x3534a03)
#16 0x0000555d58ae2f96 llvm::detail::PassModel<llvm::Function, llvm::LoopLoadEliminationPass, llvm::PreservedAnalyses, llvm::AnalysisManager<llvm::Function>>::run(llvm::Function&, llvm::AnalysisManager<llvm::Function>&) (/opt/compiler-explorer/clang-assertions-trunk/bin/opt+0x3b71f96)
#17 0x0000555d57e8991a llvm::PassManager<llvm::Function, llvm::AnalysisManager<llvm::Function>>::run(llvm::Function&, llvm::AnalysisManager<llvm::Function>&) (/opt/compiler-explorer/clang-assertions-trunk/bin/opt+0x2f1891a)
#18 0x0000555d55fd9c76 llvm::detail::PassModel<llvm::Function, llvm::PassManager<llvm::Function, llvm::AnalysisManager<llvm::Function>>, llvm::PreservedAnalyses, llvm::AnalysisManager<llvm::Function>>::run(llvm::Function&, llvm::AnalysisManager<llvm::Function>&) (/opt/compiler-explorer/clang-assertions-trunk/bin/opt+0x1068c76)
#19 0x0000555d57e88486 llvm::ModuleToFunctionPassAdaptor::run(llvm::Module&, llvm::AnalysisManager<llvm::Module>&) (/opt/compiler-explorer/clang-assertions-trunk/bin/opt+0x2f17486)
#20 0x0000555d55fda676 llvm::detail::PassModel<llvm::Module, llvm::ModuleToFunctionPassAdaptor, llvm::PreservedAnalyses, llvm::AnalysisManager<llvm::Module>>::run(llvm::Module&, llvm::AnalysisManager<llvm::Module>&) (/opt/compiler-explorer/clang-assertions-trunk/bin/opt+0x1069676)
#21 0x0000555d57e8652e llvm::PassManager<llvm::Module, llvm::AnalysisManager<llvm::Module>>::run(llvm::Module&, llvm::AnalysisManager<llvm::Module>&) (/opt/compiler-explorer/clang-assertions-trunk/bin/opt+0x2f1552e)
#22 0x0000555d55bb5068 llvm::runPassPipeline(llvm::StringRef, llvm::Module&, llvm::TargetMachine*, llvm::TargetLibraryInfoImpl*, llvm::ToolOutputFile*, llvm::ToolOutputFile*, llvm::ToolOutputFile*, llvm::StringRef, llvm::ArrayRef<llvm::StringRef>, llvm::ArrayRef<llvm::PassPlugin>, llvm::opt_tool::OutputKind, llvm::opt_tool::VerifierKind, bool, bool, bool, bool, bool, bool) (/opt/compiler-explorer/clang-assertions-trunk/bin/opt+0xc44068)
#23 0x0000555d55ad9cca main (/opt/compiler-explorer/clang-assertions-trunk/bin/opt+0xb68cca)
#24 0x00007fa6fb99f083 __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x24083)
#25 0x0000555d55ba7a1e _start (/opt/compiler-explorer/clang-assertions-trunk/bin/opt+0xc36a1e)
Compiler returned: 139
```

**Comment by efriedma-quic** [comment]
CC @huntergr-arm @david-arm @fhahn

**Comment by efriedma-quic** [comment]
https://reviews.llvm.org/D133020 is under review

**Comment by efriedma-quic** [comment]
Fixed by [D133020](https://reviews.llvm.org/D133020)
